tests/test_adaptor_strict.cc

```diff
@@ -27,7 +27,9 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include "tests.hh"
+#include <boost/test/unit_test.hpp>
+#include <boost/mpl/list.hpp>
+
```

Comment:
Should _our_ headers not come before the system headers?

Reply:
[google code style](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes) (on which this project is based, right?) is 

>  Related header, C system headers, C++ standard library headers, other libraries' headers, your project's headers.

Reply:
I think it makes sense: start from the most generic to the more specific. For examples, `<cstddef>` defines size_t/std::size_t, so I would include it before using these types, even in headers from the same project.